### PR TITLE
Allow for custom export file name

### LINKED ISSUE #1417: Pipeline export: allow for custom export file name
**Is your feature request related to a problem? Please describe.**

Currently export produces a file with the pipeline name, e.g. `my-pipeline.py` or `my-pipeline.yaml` for Kubeflow Pipelines environments. With the newly added support for Apache Airflow as a runtime environment in version 2.1, export also produces a `my-pipeline.py` file when AA is selected as target environment. (The content of `my-pipeline.py` (KFP) and `my-pipeline.py` (AA) is very different.)

**Describe the solution you'd like**
To make it easier for users to determine what the purpose of a generated file is, Elyra should
- generate a default file name that identifies the export target, e.g. `my-pipeline-kfp-dsl.py`
- allow the user to choose a custom name prior to processing the export request

## Update 
Since each runtime type now only has a single export format -- `.py` for Airflow and `.yaml` for KFP) -- it's probably sufficient to allow the user to customize the export file name and have the default file name pre-populated as the pipeline file name. This behavior is similar to that which already exists for pipeline run/submission.

### PR DESCRIPTION
**Pull request by salonee13**
Fixes #1417 

### What changes were proposed in this pull request?
Added functionality that allows the user to change the export file name to their own custom name if needed, if not, keeps it as the export path name

This shows the default export name: 
export_pt1

This shows that it allows you to customize the export name to whatever you want:  
export_pt2

This shows that the file exports and saves successfully with the custom export name:  
export_pt3

### How was this pull request tested?
Tested manually, see screenshots shown above. After the integration tests run, we will add a test to make sure this functionality works. 

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Review by kiersten-stokes — COMMENTED** [review]
This is looking really great!

Could we add some space beneath the export name field to bring it in like with how it looks in the submission dialog (shown below)?

Screen Shot 2022-11-07 at 2 26 42 PM

File: packages/pipeline-editor/src/PipelineExportDialog.tsx [hunk 1]
```diff
@@ -49,15 +49,26 @@ interface IProps {
...
     <form className="elyra-dialog-form">
+      <label htmlFor="export_name">Export Name:</label>
+      <br />
+      <input
+        type="text"
+        id="export_name"
+        name="export_name"
+        defaultValue={exportName}
+        data-form-required
+      />
```
Comment (karlaspuldaro) [inline packages/pipeline-editor/src/PipelineExportDialog.tsx:71]: Can you add one or two more break lines after this input element, so it gives some spacing between that and the next element below?

**Comment by karlaspuldaro** [comment]
Thank you @salonee13 for implementing this enhancement, it looks great and works as expected :)
I only had one comment regarding the styling but everything else looks good to me!

For the integration test, you can add a new file (maybe call it `pipelineexport.ts`) to the [integration tests folder](https://github.com/elyra-ai/elyra/tree/main/tests/integration).
One test case should be sufficient, and the workflow/commands would look something like:
1. create a runtime config
2. open a pipeline test file (`helloworld.pipeline`)
3. click export button
4. edit export name to something else
5. select the expected runtime config
6. press ok
7. check export name is in the `succeeded` dialog
8. press ok o the `succeeded` dialog
9. check the new file is listed in the file browser

**Comment by ptitzler** [comment]
Could you also change the order in which information is rendered and rename the label to `Export Filename`?

image

By moving the filename input to the bottom we group related information: the first inputs are specific to the runtime configuration that will be applied and the last three are specific to the exported file that will be produced:
 - choose a file format
 - choose a filename
 - choose whether to overwrite the file if it already exists

File: packages/pipeline-editor/src/PipelineExportDialog.tsx (hunk 1 above)
Comment (salonee13) [reply packages/pipeline-editor/src/PipelineExportDialog.tsx]: Made the changes, and added two break lines.

**Review by kiersten-stokes — APPROVED** [review]
I'm approving as-is because we decided offline to add the above-mentioned integration test a follow-up issue/PR

Thank you so much, Salonee! I'll be getting a lot of use out of this feature

**Review by karlaspuldaro — APPROVED** [review]

**Pull request merged**